### ISSUE TITLE
Building as described in README fails

### ISSUE THREAD

**Issue by Saikrishna-Achalla** [sga001/sealpir-rust#3]
<img width="947" alt="image" src="https://github.com/user-attachments/assets/ab1cae03-6c94-457d-9306-b3e7b9e974e8">
<img width="1884" alt="image" src="https://github.com/user-attachments/assets/dc8863f0-e219-43a9-a73a-f69895b94071">
Attached above are the images for the build error I get. Ran the above in a fresh EC2 instance and still failed. Any help would be appreciated!

**Comment by sga001** [comment]
It's been a really long time since I've looked into this. Nobody really uses this version of SealPIR. Getting Rust and C++ to be happy is really hard, especially as compilers keep changing. 

If you used the exact same version of gcc and rust, maybe there is some hope. Otherwise, it seems like it'll be a pain.

I suggest that you either:

(1) Use the official version of SealPIR (C++): https://github.com/Microsoft/SealPIR

(2) Use Apple's version of SealPIR (Swift): https://github.com/apple/swift-homomorphic-encryption/blob/main/Sources/PrivateInformationRetrieval/IndexPirProtocol.swift#L23

(3) Try Tancrede's version of SealPIR (Rust): https://github.com/tlepoint/fhe.rs/blob/c02cde75a52437bc23f49ebfda084ec152b61248/crates/fhe/examples/sealpir.rs#L4

**Comment by Saikrishna-Achalla** [comment]
Thank you for the response. I was mainly asking about this as I'm interested in looking into the mPIR work and that relies on this version of sealPIR. Would pointing the mPIR cargo file to one of the versions of SealPIR above work instead?

**Comment by sga001** [comment]
No, because that version is C++, not Rust.

When you tried compiling the code in this repo, were you using the exact same gcc version and rust version that we tested with?

gcc/g++ version 11.3
Rust version 1.66

?

**Comment by Saikrishna-Achalla** [comment]
Yes, I used the same gcc and rust versions you used but the build still fails
